internal/services/apimanagement/api_management_policy_fragment_resource.go

```diff
@@ -0,0 +1,232 @@
+		}),
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Create: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Read:   pluginsdk.DefaultTimeout(5 * time.Minute),
+			Update: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Delete: pluginsdk.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*pluginsdk.Schema{
+			"name": schemaz.SchemaApiManagementChildName(),
+
+			"resource_group_name": commonschema.ResourceGroupName(),
+
+			"api_management_name": schemaz.SchemaApiManagementName(),
+
+			"format": {
+				Type:     pluginsdk.TypeString,
+				ForceNew: true,
+				Optional: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(policyfragment.PolicyFragmentContentFormatRawxml),
+					string(policyfragment.PolicyFragmentContentFormatXml),
+				}, false),
+				Default: policyfragment.PolicyFragmentContentFormatXml,
+			},
+
+			"value": {
+				Type:             pluginsdk.TypeString,
+				Required:         true,
+				DiffSuppressFunc: XmlWithDotNetInterpolationsDiffSuppress,
+			},
+
+			"description": {
+				Type:     pluginsdk.TypeString,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func resourceApiManagementPolicyFragmentCreate(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).ApiManagement.PolicyFragmentClient
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	subscriptionId := meta.(*clients.Client).Account.SubscriptionId
+	defer cancel()
+
+	id := policyfragment.NewPolicyFragmentID(subscriptionId, d.Get("resource_group_name").(string), d.Get("api_management_name").(string), d.Get("name").(string))
+	format := policyfragment.PolicyFragmentContentFormat(d.Get("format").(string))
+	if d.IsNewResource() {
+		existing, err := client.Get(ctx, id, policyfragment.GetOperationOptions{
+			Format: &format,
+		})
+		if err != nil {
+			if !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for presence of existing %s: %s", id, err)
+			}
+		}
+
+		if !response.WasNotFound(existing.HttpResponse) {
+			return tf.ImportAsExistsError("azurerm_api_management_policy_fragment", id.ID())
+		}
+	}
+
+	description := d.Get("description").(string)
+	value := d.Get("value").(string)
+
+	parameters := policyfragment.PolicyFragmentContract{
+		Properties: &policyfragment.PolicyFragmentContractProperties{
+			Description: pointer.To(description),
+			Format:      pointer.To(format),
+			Value:       value,
+		},
+	}
+
+	if err := client.CreateOrUpdateThenPoll(ctx, id, parameters, policyfragment.CreateOrUpdateOperationOptions{}); err != nil {
+		return fmt.Errorf("creating/updating %s: %+v", id, err)
+	}
+
+	d.SetId(id.ID())
+
+	return resourceApiManagementPolicyFragmentRead(d, meta)
+}
+
+func resourceApiManagementPolicyFragmentUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).ApiManagement.PolicyFragmentClient
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
```

Comment:
since this is only for the Update method we can make this:

```suggestion
	ctx, cancel := timeouts.ForUpdate(meta.(*clients.Client).StopContext, d)
```